src/Aquifer.Public.API/Endpoints/ParentResources/List/Request.cs

```diff
@@ -0,0 +1,14 @@
+namespace Aquifer.Public.API.Endpoints.ParentResources.List;
+
+public sealed class Request
+{
+    /// <summary>
+    /// Optional language to search on by internal Aquifer language id. If no language is specified then English will be used by default. Only one of language id or code may be provided.
+    /// </summary>
+    public int? LanguageId { get; init; }
```

Comment:
I could see users wanting to know all languages a collection is available for. I would say that if no language id/code is provided, rather than default to English, it returns all. And to that end, I'd let them send a list of language ids. My thinking is that the response would have the collection name and information, and then a list of `AvailableLanguages` with an object that's like `[{"languageId": 2, "languageCode": "TPI", "resourceItemCount": 153}, etc.]`

Reply:
> I could see users wanting to know all languages a collection is available for.

That's a different use case than the [route that I copied](https://github.com/BiblioNexusStudio/aquifer-server/blob/master/src/Aquifer.API/Endpoints/Resources/ParentResources/List/Endpoint.cs) from the internal API.  That route is language specific and only returns the localized name, not the original collection's name, unless that name happens to be the `code` or in the license information.  That existing route (and thus the route in this PR as I originally implemented it) was designed to show information for only a single language.

>  I would say that if no language id/code is provided, rather than default to English, it returns all.

It's not currently defaulting to English but will throw an error if no language information is provided in the request.

> I'd let them send a list of language ids.

As it stands now they could just hit the route multiple times, once for each language they care about.  I have concerns about returning more than one language in a single response (see below).

> My thinking is that the response would have the collection name and information, and then a list of `AvailableLanguages` with an object that's like `[{"languageId": 2, "languageCode": "TPI", "resourceItemCount": 153}, etc.]`

Fetching the counts for a single language is already pushing what I'd consider the maximum DB performance threshold (~300 ms) and it's only going to get more expensive to count these over time as we fill the Aquifer.  Thus, we'd probably want to leave off the counts if we accept more than one `languageId` or find another solution for the counts like keeping a table up-to-date with total counts or using a view that keeps the counts up-to-date automagically.

So all that said, I'm wondering if the best thing here might be two different routes, one that is "by language" (i.e. this PR's route) and another new route that returns more of a summary view which shows all collections in our system and the languages for which they have any translated content (but without counts).  Thoughts?

Reply:
I'm actually not likely my idea in the last comment because it looks like the counts _are_ necessary in order to make sense of the data.  Check out the prod data for Malaysian where there are only two collections with any content but we've localized all of the collection names in advance.  Looking at current data, a summary route that only relies on the presence of a localized name will return a summary for every collection for every language.
![image](https://github.com/user-attachments/assets/77e1b6c1-053d-4e85-931b-b005457ebac1).

Reply:
Update after video chat: There will be two routes, List and Get (by code).  List will have minimal data from the english parent resource.  Get will join on localization data and include counts for all languages but for only a single parent resource.